src/components/v5/common/ActionSidebar/partials/forms/ArbitraryTxsForm/partials/ArbitraryTransactionsTable/hooks.tsx

```diff
@@ -0,0 +1,90 @@
+import { createColumnHelper, type ColumnDef } from '@tanstack/react-table';
+import { isAddress } from 'ethers/lib/utils';
+import React, { useMemo } from 'react';
+
+import { useMobile } from '~hooks/index.ts';
+import getMaskedAddress from '~shared/MaskedAddress/getMaskedAddress.ts';
+import { formatText } from '~utils/intl.ts';
+import { type AddTransactionTableModel } from '~v5/common/ActionSidebar/partials/forms/ArbitraryTxsForm/types.ts';
+import UserAvatar from '~v5/shared/UserAvatar/UserAvatar.tsx';
+
+import CellDescription from './CellDescription.tsx';
+
+export const useArbitraryTxsTableColumns = (): ColumnDef<
+  AddTransactionTableModel,
+  string
+>[] => {
+  const columnHelper = useMemo(
+    () => createColumnHelper<AddTransactionTableModel>(),
+    [],
+  );
+  const isMobile = useMobile();
+
+  const columns: ColumnDef<AddTransactionTableModel, string>[] = useMemo(
+    () => [
+      columnHelper.accessor('contractAddress', {
+        enableSorting: false,
+        header: () => (
+          <span className="text-sm text-gray-600">
+            {formatText({ id: 'table.row.contract' })}
+          </span>
+        ),
+        cellContentWrapperClassName: '!justify-start',
+
+        cell: ({ getValue }) => {
+          const address = getValue();
+          const isAddressValid = isAddress(address);
+          const maskedAddress = getMaskedAddress({
+            address,
+          });
+          return (
+            <span className="flex max-w-full items-center gap-2">
+              {isAddressValid && <UserAvatar userAddress={address} size={20} />}
```

Comment:
Do you think we could use `AvatarWithAddress` here? not sure if it would be too much work, but it seems like a nifty component :100: